### PR TITLE
fix HttpObjectAggregator content length error causing the pipeline to be unable to handle data

### PR DESCRIPTION
**Pull request by Halooooooo**
Motivation:

When using HttpObjectAggregator, Http Response returns an illegal Content-Length, HttpObjectAggregator will wait for channel inactive，when the read is complete, the channelpipline will not continue, it should trigger an exception
Modification:

In the channelInactive of MessageAggregator, add aggregating judgment. When aggregating is not completed, and channel inactive, an exception should be thrown to let the channelHandler in the channelpipline handle it.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@Halooooooo can you please provide a unit test that shows the problem ?

File: codec/src/main/java/io/netty/handler/codec/MessageAggregator.java [hunk 1]
```diff
@@ -431,6 +431,9 @@ public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
 
     @Override
     public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+        if(this.aggregating){
+            super.exceptionCaught(ctx,new RemainLengthNotReadException("expected to aggregate more data, currently "+this.currentMessage.content().capacity()+" bits are accepted, but the channel has been closed. "));
```
Comment (hyperxpro) [inline codec/src/main/java/io/netty/handler/codec/MessageAggregator.java:435]: The maximum allowed line length is 120. Trim it down, please.

Comment (hyperxpro) [inline codec/src/main/java/io/netty/handler/codec/MessageAggregator.java:435]:
```suggestion
            ctx.fireExceptionCaught(ctx,new RemainLengthNotReadException("expected to aggregate more data, currently "+this.currentMessage.content().capacity()+" bits are accepted, but the channel has been closed. "));
```

**Comment by Halooooooo** [comment]
> @Halooooooo can you please provide a unit test that shows the problem ?

Sorry for not replying in time，
This is a httpserver returning wrong content-length：


```
import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;
import com.sun.net.httpserver.HttpServer;
import java.io.IOException;
import java.io.OutputStream;
import java.net.InetSocketAddress;
import java.nio.charset.Charset;
public class httpserver {
    public static final String response = """
            {
             "中文测试":"中文测试",
             "vcbppwdbh": [
              "yM4EGF",
              false
             ],
             "lnjwzwo": false,
             "pvewxw": [
              {
               "iptwgrjiy": {
                "kdytbntsfhsb": [
                 "W6Gpfp9eVVejf",
                 true,
                 "ToMdSUrkxS0_",
                 [
                  "rTRYKIvEkV47bK",
                  "QXtlTny",
                  1315000511.5859537,
                  "U6F6",
                  true
                 ],
                 "V"
                ],
                "tgjkryfv": false,
                "qjweuo": true,
                "xzqub": {
                 "wmkqatsul": "oGiLWkoB6-8ETEc0e",
                 "bwjqgj": -243242530,
                 "ifemj": true,
                 "vmflkdmq": "EwxflPbCeMSPjEm",
                 "ihshbgvjiuep": true,
                 "huoodd": [
                  105801123,
                  1974513878.0048652,
                  true
                 ]
                },
                "sncmji": 1798784983.5633993,
                "tickgz": "Qz77nVaiKeVKFg-"
               }
              },
              false,
              -1793514816.215511
             ]
            }
            """;


    public static void main(String[] arg) throws Exception {
        HttpServer server = HttpServer.create(new InetSocketAddress(8000), 0);
        server.createContext("/contentLength/normal", new ContentLengthNormal());
        server.createContext("/contentLength/greater", new ContentLengthGreater());
        server.createContext("/contentLength/less", new ContentLengthLess());
        server.start();
    }


    // This class is used to return ContentLength greater than the normal value
    static class ContentLengthGreater implements HttpHandler {
        @Override
        public void handle(HttpExchange exchange) throws IOException {
            byte[] gbks = response.getBytes(Charset.forName("GBK"));
            exchange.getResponseHeaders().set("Content-Type","application/json; charset=gbk");
            exchange.sendResponseHeaders(200, gbks.length+100);
            OutputStream os = exchange.getResponseBody();
            os.write(gbks);
            os.close();
        }
    }
    // This class is used to return ContentLength less than the normal value

    static class ContentLengthLess implements HttpHandler {
        @Override
        public void handle(HttpExchange exchange) throws IOException {
            byte[] gbks = response.getBytes(Charset.forName("GBK"));
            exchange.getResponseHeaders().set("Content-Type","application/json; charset=gbk");
            exchange.sendResponseHeaders(200, gbks.length-100);
            OutputStream os = exchange.getResponseBody();
            os.write(gbks);
            os.close();
        }
    }

    // This class is used to return Content-Length normal value
    static class ContentLengthNormal implements HttpHandler {
        @Override
        public void handle(HttpExchange exchange) throws IOException {
            byte[] gbks = response.getBytes(Charset.forName("GBK"));
            exchange.getResponseHeaders().set("Content-Type","application/json; charset=gbk");
            exchange.sendResponseHeaders(200, gbks.length);
            OutputStream os = exchange.getResponseBody();
            os.write(gbks);
            os.close();
        }
    }
}
```
this is netty bootstrap:
```

import io.netty.bootstrap.Bootstrap;
import io.netty.channel.*;
import io.netty.channel.nio.NioEventLoopGroup;
import io.netty.channel.socket.SocketChannel;
import io.netty.channel.socket.nio.NioSocketChannel;
import io.netty.handler.codec.http.*;
import io.netty.handler.codec.http.cookie.ClientCookieEncoder;
import io.netty.handler.codec.http.cookie.DefaultCookie;
import io.netty.handler.logging.LogLevel;
import io.netty.handler.logging.LoggingHandler;
import io.netty.util.ReferenceCountUtil;

import java.net.URI;

/**
 * A simple HTTP client that prints out the content of the HTTP response to
 * {@link System#out} to test {@link HttpSnoopServer}.
 */
public final class HttpClient {

    static final String URL = System.getProperty("url", "http://127.0.0.1:8001/contentLength/greater");

    public static void main(String[] args) throws Exception {
        URI uri = new URI(URL);
        String scheme = uri.getScheme() == null? "http" : uri.getScheme();
        String host = uri.getHost() == null? "127.0.0.1" : uri.getHost();
        int port = uri.getPort();


        // Configure the client.
        EventLoopGroup group = new NioEventLoopGroup();
        try {
            Bootstrap b = new Bootstrap();
            b.group(group)
             .channel(NioSocketChannel.class)
             .handler(new ChannelInitializer<SocketChannel>() {
                 @Override
                 protected void initChannel(SocketChannel ch) throws Exception {
                     SocketChannel channel = (SocketChannel) ch;
                     channel.config().setKeepAlive(true);
                     channel.config().setTcpNoDelay(true);
                     channel.pipeline()
                             .addFirst(new LoggingHandler(LogLevel.DEBUG));
                     channel.pipeline()
                             .addLast("httpDecoder",new HttpResponseDecoder(409600,81920,81920+409600))
                             .addLast("aggregator",new HttpObjectAggregator(65536))
                             .addLast("httpEncoder",new HttpRequestEncoder());
                     channel.pipeline().addLast(new ChannelDuplexHandler(){
                         @Override
                         public void channelRead(final ChannelHandlerContext ctx, final Object msg) {
                             try {
                                 if (msg instanceof FullHttpResponse ) {
                                     System.out.println("get msg");
                                 }
                             }finally {
                                 ReferenceCountUtil.release(msg);
                             }
                         }
                     });
                 }
             });

            // Make the connection attempt.
            Channel ch = b.connect(host, port).sync().channel();

            // Prepare the HTTP request.
            HttpRequest request = new DefaultFullHttpRequest(
                    HttpVersion.HTTP_1_1, HttpMethod.POST, uri.getRawPath());
            request.headers().set(HttpHeaderNames.HOST, host);
            request.headers().set(HttpHeaderNames.CONNECTION, HttpHeaderValues.CLOSE);
            request.headers().set(HttpHeaderNames.ACCEPT_ENCODING, HttpHeaderValues.GZIP);

            // Set some example cookies.
            request.headers().set(
                    HttpHeaderNames.COOKIE,
                    ClientCookieEncoder.STRICT.encode(
                            new DefaultCookie("my-cookie", "foo"),
                            new DefaultCookie("another-cookie", "bar")));

            // Send the HTTP request.
            ch.writeAndFlush(request);

            // Wait for the server to close the connection.
            ch.closeFuture().sync();
        } finally {
            // Shut down executor threads to exit.
            group.shutdownGracefully();
        }
    }
}
```
When aggregating is not completed, and channel inactive

**Pull request closed without merging**